test/test_sparse_csr.py

```diff
@@ -2045,8 +2045,8 @@ def _test_spadd_shape(fn, nnz, shape):
             res_sparse_sparse = fn(y, x)
             res_dense_sparse = fn(y.to_dense(), x)
             res_sparse_dense = fn(y, x.to_dense())
-            expected = fn(y.to_dense(), x.to_dense()).to_sparse_csr()
-            self.assertEqual(res_sparse_sparse, expected)
+            expected = fn(y.to_dense(), x.to_dense())
+            self.assertEqual(res_sparse_sparse.to_dense(), expected)
```

Comment:
When `expected` is strided, `assertEqual` converts the other argument to strided as well, so:
```suggestion
            self.assertEqual(res_sparse_sparse, expected)
```